presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergMetadata.java

```diff
@@ -662,4 +702,335 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
+        buildProps.put(STORAGE_TABLE, storageTableName);
+        buildProps.put(PRESTO_VIEW_FLAG, "true");
+        if (replace && definition.getComment().isPresent()) {
+            buildProps.put(TABLE_COMMENT, definition.getComment().get());
+        }
+        else {
+            // Add a default table comment when the view is being created
+            buildProps.put(TABLE_COMMENT, "Presto Materialized View");
+        }
+        properties = ImmutableMap.<String, String>builder()
+            .putAll(buildProps)
+            .build();
+
+        io.prestosql.plugin.hive.metastore.Column dummyColumn = new io.prestosql.plugin.hive.metastore.Column("dummy", HIVE_STRING, Optional.empty());
+
+        String schemaName = viewName.getSchemaName();
+        String tableName = viewName.getTableName();
+        Table.Builder tableBuilder = Table.builder()
+                .setDatabaseName(schemaName)
+                .setTableName(tableName)
+                .setOwner(session.getUser())
+                .setTableType(org.apache.hadoop.hive.metastore.TableType.VIRTUAL_VIEW.name())
+                .setDataColumns(ImmutableList.of(dummyColumn))
+                .setPartitionColumns(ImmutableList.of())
+                .setParameters(properties);
+        tableBuilder.getStorageBuilder()
+                .setStorageFormat(VIEW_STORAGE_FORMAT)
+                .setLocation("");
+
+        HiveIdentity identity = new HiveIdentity(session);
+        Optional<Table> existing = metastore.getTable(identity, viewName.getSchemaName(), viewName.getTableName());
+
+        // It's a create command where the materialized view already exists and 'if not exists' clause is specified
+        if (!replace && existing.isPresent() && !ignoreExisting) {
+            throw new TableAlreadyExistsException(viewName);
+        }
+
+        List<ColumnMetadata> columns = new ArrayList<>();
+        for (ConnectorMaterializedViewDefinition.Column col : definition.getColumns()) {
+            columns.add(new ColumnMetadata(col.getName(), typeManager.getType(col.getType())));
+        }
+
+        Map<String, Object> finalProperties = new HashMap<>();
+        finalProperties.put(FILE_FORMAT_PROPERTY, FileFormat.PARQUET);
+        if (definition.getProperties().containsKey(PARTITIONING_PROPERTY)) {
+            finalProperties.put(PARTITIONING_PROPERTY, definition.getProperties().get(PARTITIONING_PROPERTY));
+        }
+
+        SchemaTableName storageTable = new SchemaTableName(viewName.getSchemaName(), storageTableName);
+        ConnectorTableMetadata tableMetadata = new ConnectorTableMetadata(storageTable, columns, finalProperties, Optional.empty());
+
+        Optional<ConnectorNewTableLayout> layout = getNewTableLayout(session, tableMetadata);
+        finishCreateTable(session, beginCreateTable(session, tableMetadata, layout), ImmutableList.of(), ImmutableList.of());
+
+        tableBuilder.setViewOriginalText(Optional.of(encodeMaterializedViewData(definition)))
```

Comment:
If we move the entire `Table.Builder` part down here, we could directly chain the whole thing
```java
Table table = Table.builder()
        ...
        .build();

Reply:
Yes. The function has now been restructured and takes care of this comment.